torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -554,6 +554,23 @@ void Reducer::delay_all_reduce() {
     }
   }
 
+  // To avoid confusion around why static graph is picking up
+  // some parameters as unused on a rank vs not, we log
+  // unused parameter names for each rank for better
+  // debugability
+  if (ddp_debug_level_ == c10d::DebugLevel::Info ||
+      ddp_debug_level_ == c10d::DebugLevel::Detail) {
+    for (const auto& unused_index : unused_parameters_) {
+      auto param_name = param_names_.find(unused_index);
+      TORCH_INTERNAL_ASSERT(
+          param_name != param_names_.end(),
+          "Expected to find parameter name from unused parameters map in debug mode.");
+      LOG(INFO) << "[Rank " << process_group_->getRank() << "]: "
```

Comment:
actually, could we just make it one big sentence, if there are 128 ranks and 100 unused params, this will result in 12.8k log lines, instead of just 128.

Could we just accumulate the unused param names in a container, string join them, and log a single time?

Reply:
Yes, will do.

Reply:
New output format added to make sure the log is a single line.